cartography/intel/crxcavator.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import json
+import requests
+import requests.auth
+import os
+
+from cartography.util import run_cleanup_job
+
+logger = logging.getLogger(__name__)
+
+# API key for CRXcavator generaated from crxcavator.io portal
+CRXcavator_API_KEY = os.environ.get('CREDENTIALS_CRXCAVATOR_API_KEY')
+
+# API for the CRXcavator API - https://api.crxcavator.io/v1 as of 07/09/19
+CRXcavator_API_BASE_URL = os.environ.get('CRXCAVATOR_URL')
+
+
+def get_extensions():
+    """
+    Get all of the installed extension metadata
+    :return: JSON text blob containing all extension metadata defined at
+    https://crxcavator.io/apidocs#tag/group/paths/~1group~1extensions~1combined/get
+    """
+    return call_crxcavator_api("/group/extensions/combined")
+
+
+def get_users_extensions():
+    """
+    Gets listing of all users who have installed each extension
+    :return: JSON text blob containing user email to extension id mapping defined at
+    https://crxcavator.io/apidocs#tag/group/paths/~1group~1users~1extensions/get
+    """
+    return call_crxcavator_api("/group/users/extensions")
+
+
+def call_crxcavator_api(api_and_parameters):
+    """
+    Perform the call requested to the CRXcavator API
+    :param api_and_parameters: Query string for the API including the required parameters
+    :return: Returns JSON text blob for the API called. API spec is at https://crxcavator.io/apidocs
+    """
+    uri = CRXcavator_API_BASE_URL + api_and_parameters
+    data = requests.get(
+        uri,
+        headers={'Accept': 'application/json',
+                 'API-Key': CRXcavator_API_KEY})
+    if data.ok:
+        return data.json()
+    else:
+        logger.error('Error during CRXcavator API call {}'.format(data.status_code))
+        return None
+
+
+def ingest_extensions(session, update_tag):
```

Comment:
Please refactor this to the following pattern:

`get_extension()` - to get the data (seems you already have this)
`transform_extension()` - to perform data adjustments (if needed) on the returned json data from `get_extension()`, for example it looks like handling permission warnings below performs some data massaging
`load_extension()` - to actually load the transformed data into neo4j

Then, define a function called `sync_extension` to call `get`, `transform`, and `load`.

Reply:
refactored to follow get,transform,load methodology for extensions and users